src/field_theory/separable.lean

```diff
@@ -449,6 +450,47 @@ begin
   exact_mod_cast (enat.add_one_le_of_lt hq)
 end
 
+lemma separable.squarefree {p : polynomial F}  (hsep : separable p) : squarefree p :=
+begin
+  rw multiplicity.squarefree_iff_multiplicity_le_one p,
+  intro f,
+  by_cases hunit : is_unit f,
+  { exact or.inr hunit },
+  exact or.inl (multiplicity_le_one_of_separable hunit hsep)
+end
+
+/--If `n ≠ 0` in `F`, then ` X ^ n - a` is separable for any `a ≠ 0`. -/
+lemma separable_of_X_pow_sub_C {n : ℕ} (a : F) (hn : ↑n ≠ (0 : F)) (ha : a ≠ 0) :
+  separable (X ^ n - C a) :=
+begin
+  cases nat.eq_zero_or_pos n with hzero hpos,
+  { exfalso,
+    rw hzero at hn,
+    exact hn (refl 0) },
+  apply (separable_def' (X ^ n - C a)).2,
+  use [-C (a⁻¹), (C ((a⁻¹) * (↑n)⁻¹) *  X)],
+  rw [derivative_sub, derivative_C, sub_zero, derivative_pow X n, derivative_X, mul_one],
+  calc -C a⁻¹ * (X ^ n - C a) + C (a⁻¹ * (↑n)⁻¹) * X * (↑n * X ^ (n - 1))
+      = -C a⁻¹ * (X ^ n - C a) + C (a⁻¹ * (↑n)⁻¹) * (↑n * (X * X ^ (n - 1))) : by ring
+  ... = -C a⁻¹ * (X ^ n - C a) + C (a⁻¹ * (↑n)⁻¹) * (↑n * (X ^ n)) : by rw [←pow_succ,
+  ← nat.pred_eq_sub_one, ← nat.succ_eq_add_one, nat.succ_pred_eq_of_pos hpos]
```

Comment:
This should be indented here to better distinguish the continuation from new `calc` steps:
```suggestion
    ← nat.pred_eq_sub_one, ← nat.succ_eq_add_one, nat.succ_pred_eq_of_pos hpos]
```
However, I'm surprised this step takes 4 rewrites. Can this be extracted into a lemma?

Reply:
It was like that, there was
```
  have hpred : n - 1 + 1 = n,
  { exact nat.succ_pred_eq_of_pos hpos },
```
and then `by rw [←pow_succ, hpred]` was enough. Note that if I write just `by rw [←pow_succ, nat.succ_pred_eq_of_pos hpos]`. I can of course go back to the previous version.

Reply:
Sorry, I meant a standalone lemma stating that `X * X ^ (n - 1) = X ^ n` as this seems like a common manipulation.

Reply:
Ah yes, I can do it, but I see that there is `pow_mul_pow_sub`, maybe is it better to just use it?

Reply:
Oh yeah, I think that would be better.